### PR TITLE
Client tls policy

### PR DESCRIPTION
**Pull request by maphad**
Add configuration for traffic director clientTlsPolicy resource

part of https://github.com/hashicorp/terraform-provider-google/issues/14040

If this PR is for Terraform, I acknowledge that I have:

- [ x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_networksecurity_client_tls_policy`
```

### CODE REVIEW

**Comment by shuyama1** [comment]
Reassigning this PR as it's still WIP and I will be out for the next week.
/gcbrun

File: mmv1/products/networksecurity/ClientTlsPolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+name: 'ClientTlsPolicy'
+base_url: 'projects/{{project}}/locations/{{location}}/clientTlsPolicies'
+create_url: 'projects/{{project}}/locations/{{location}}/clientTlsPolicies?clientTlsPolicyId={{name}}'
+min_version: beta
+update_verb: :PATCH
+update_mask: true
+description: |
+  ClientTlsPolicy is a resource that specifies how a client should authenticate connections to backends of a service. This resource itself does not affect configuration unless it is attached to a backend service resource.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: 'https://cloud.google.com/traffic-director/docs/reference/network-security/rest/v1beta1/projects.locations.clientTlsPolicies'
```
Comment (zli82016) [inline mmv1/products/networksecurity/ClientTlsPolicy.yaml:24]: Is the user guild available to add to the resource?

File: mmv1/products/networksecurity/ClientTlsPolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,160 @@
+# Copyright 2023 Google Inc.
```
Comment (zli82016) [inline mmv1/products/networksecurity/ClientTlsPolicy.yaml:1]: This file should be put into the folder products/networkservices.

Comment (maphad) [reply mmv1/products/networksecurity/ClientTlsPolicy.yaml:1]: Why is that? This is part of the networksecurity API.

File: mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+	clientTlsPolicyName := fmt.Sprintf("tf-test-client-tls-policy-%s", RandString(t, 10))
+
+	VcrTest(t, resource.TestCase{
+		PreCheck:     func() { AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy: testAccCheckNetworkSecurityClientTlsPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+        Config: testAccNetworkSecurityClientTlsPolicy_basic(clientTlsPolicyName),
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb:23]: Indentation is weird here.

File: mmv1/templates/terraform/examples/network_security_client_tls_policy_advanced.tf.erb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  labels                 = {
+    foo = "bar"
+  }
+  description            = "my description"
+  client_certificate {
+    certificate_provider_instance {
+        plugin_instance = "google_cloud_private_spiffe"
+      }
+    }
+  server_validation_ca {
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/network_security_client_tls_policy_advanced.tf.erb:13]: As serverValidationCa is an array, can you please add more server_validation_ca blocks here? Thanks.

File: mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+<% unless version == 'ga' -%>
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccNetworkSecurityClientTlsPolicy_update(t *testing.T) {
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb:12]: Can you please add the updating tests for other fields, too? Thanks.

File: mmv1/templates/terraform/examples/network_security_client_tls_policy_basic.tf.erb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+resource "google_network_security_client_tls_policy" "<%= ctx[:primary_resource_id] %>" {
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/network_security_client_tls_policy_basic.tf.erb:1]: The resource name should be google_network_services_security_client_tls_policy as it is a product in the service networkservices.

Comment (maphad) [reply mmv1/templates/terraform/examples/network_security_client_tls_policy_basic.tf.erb:1]: The product is in network security

File: mmv1/products/networksecurity/ClientTlsPolicy.yaml (hunk 2 above)
Comment (zli82016) [reply mmv1/products/networksecurity/ClientTlsPolicy.yaml:1]: Oh. Yes, you are right. Sorry about that.

File: mmv1/templates/terraform/examples/network_security_client_tls_policy_basic.tf.erb (hunk 6 above)
Comment (zli82016) [reply mmv1/templates/terraform/examples/network_security_client_tls_policy_basic.tf.erb:1]: Oh. Yes, you are right. Sorry about that.

File: mmv1/templates/terraform/examples/network_security_client_tls_policy_advanced.tf.erb (hunk 4 above)
Comment (maphad) [reply mmv1/templates/terraform/examples/network_security_client_tls_policy_advanced.tf.erb:13]: done

File: mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb (hunk 5 above)
Comment (maphad) [reply mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb:12]: Added

File: mmv1/products/networksecurity/ClientTlsPolicy.yaml (hunk 1 above)
Comment (maphad) [reply mmv1/products/networksecurity/ClientTlsPolicy.yaml:26]: Added

File: mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb (hunk 3 above)
Comment (maphad) [reply mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb]: fixed

File: mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+		client_certificate {
+    certificate_provider_instance {
+        plugin_instance = "google_cloud_private_spiffe"
+      }
+    }
+  	server_validation_ca {
+    	grpc_endpoint {
+      	target_uri = "unix:mypath"
+    	}
+  	}
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb:60]: Can the indentation be aligned?

File: mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  	server_validation_ca {
+    	grpc_endpoint {
+      	target_uri = "unix:mypath1"
+    	}
+  	}
+		server_validation_ca {
+    	grpc_endpoint {
+      	target_uri = "unix:mypath2"
+    	}
+  	}
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb:88]: Can the indentation be aligned?

File: mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb (hunk 7 above)
Comment (maphad) [reply mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb]: Done!

File: mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb (hunk 8 above)
Comment (maphad) [reply mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb]: Done!

**Comment by zli82016** [comment]
Thanks for making the changes. Mostly LGTM. Just added comments for the indentations.

**Comment by maphad** [comment]
> Thanks for making the changes. Mostly LGTM. Just added comments for the indentations.

Thanks for review! fixed the indentation, something to do with my editor messing up when I copy. Thanks!

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**